pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -110,6 +112,28 @@ func newExecutor(im instancemgmt.InstanceManager, cfg *setting.Cfg, sessions Ses
 		sessions: sessions,
 		features: features,
 	}
+
+	cwe.getClients = func(pluginCtx backend.PluginContext, region string) (models.ClientsProvider, error) {
```

Comment:
Would suggest declaring this as a method of the executor in order to lighten the logic in `newExecutor`.

Eventually maybe some ClientGetter interface with a GetClients method could be useful if that is something we want to be able to replace. 

Understood that this will may not be the final form of the cloudWatchExecutor in future reorganization of the code.